machines/VCItemMachine/ExistingMosipVCItem/ExistingMosipVCItemMachine.ts

```diff
@@ -1066,8 +1078,19 @@ export const ExistingMosipVCItemMachine =
           },
         ),
 
-        incrementDownloadCounter: model.assign({
-          downloadCounter: ({downloadCounter}) => downloadCounter + 1,
+        incrementDownloadCounter: assign({
+          vcMetadata: context =>
+            new VCMetadata({
+              ...context.vcMetadata,
+              downloadCounter: context.vcMetadata.downloadCounter + 1,
+            }),
+        }),
+
+        deleteCounterFromVcMetadata: assign({
+          vcMetadata: context => {
+            delete context.vcMetadata.downloadCounter;
```

Comment:
We have to delete downloadCounter or set it 0?

Reply:
once the VC is completely downloaded we don't need the downloadCounter and also to keep vcMetadata as it is like previously downloadCounter is deleted.

Reply:
downloadCounter moved to context